packages/flutter/lib/src/painting/continuous_rectangle_border.dart

```diff
@@ -3,58 +3,276 @@
 // found in the LICENSE file.
 
 import 'dart:math' as math;
+import 'dart:ui' as ui;
 
 import 'basic_types.dart';
-import 'border_radius.dart';
 import 'borders.dart';
 import 'edge_insets.dart';
 
-/// A rectangular border with smooth continuous transitions between the straight
-/// sides and the rounded corners.
+/// A rectangle border with continuous corners.
+///
+/// A shape similar to a rounded rectangle, but with a smoother transition from
+/// the sides to the rounded corners.
+///
+/// The rendered shape roughly approximates that of a superellipse. In this
+/// shape, the curvature of each 90º corner around the length of the arc is
+/// approximately a gaussian curve instead of a step function as with a
+/// traditional quarter circle round. The rendered rectangle is roughly a
+/// superellipse with an n value of 5.
+///
+/// In an attempt to keep the shape of the rectangle the same regardless of its
+/// dimension (and to avoid clipping of the shape), the radius will
+/// automatically be lessened if its width or height is less than ~3x the
+/// declared radius. The new resulting radius will always be maximal in respect
+/// to the dimensions of the given rectangle.
+///
+/// The ~3 represents twice the ratio (ie. ~3/2) of a corner's declared radius
+/// and the actual height and width of pixels that are manipulated to render it.
+/// For example, if a rectangle had dimensions 80px x 100px, and a corner radius
+/// of 25, in reality ~38 pixels in each dimension would be used to render a
+/// corner and so ~76px x ~38px would be used to render both corners on a given
+/// side.
+///
+/// This shape will always have 4 linear edges and 4 90º curves. However, for
```

Comment:
Not sure I understood this part. I thought we don't do this?

Reply:
¿¿We do??

Reply:
Is this second part describing the we're-actually-clipping-down-at-the-ratio-of-2-instead-of-leaving-it-round-at-3-but-you-can't-really-tell part. I'm asking what's the rational for it.

Reply:
yup lol, that's what I'm trying to say. At small shapes where we change the ratio to 2, it looks like a stadium because of the clipping that happens at such a small screen size.

Reply:
I'm asking what's the motivation for doing something special for smaller sizes and not have one single set of rules. The users will be wondering too after reading this paragraph.